src/games/strategy/triplea/delegate/MoveValidator.java

```diff
@@ -1430,8 +1430,10 @@ public static String validateCanal(final Route route, final Collection<Unit> uni
         continue;
       }
       failureMessage = canPassThroughCanal(canalAttachment, units, player, data);
-      if (!failureMessage.isPresent()) {
```

Comment:
" I could write the whole thing on 1 long line and that doesn't make it better..."

Bit of a straw argument. It used to be a thing to write foreach loops as 3 lines in Java because the language syntax did not yet support the single line foreach loop. Anonymous inner classes used to require more lines as well.  Moving forward perhaps I'll be more concise to say statements. Moving 13 statements to 1 line clearly is not improvement, but reducing from 13 statements to 6, is a difference. 
The formatting of the code is pretty much there already. The boolean conditions can get extracted to single word helper functions.